sdk/servicebus/azure-servicebus/azure/servicebus/_pyamqp/link.py

```diff
@@ -268,5 +269,18 @@ def detach(self, close: bool = False, error: Optional[AMQPError] = None) -> None
             self._set_state(LinkState.DETACHED)
 
     def flow(self, *, link_credit: Optional[int] = None, **kwargs: Any) -> None:
-        self.current_link_credit = link_credit if link_credit is not None else self.link_credit
-        self._outgoing_flow(**kwargs)
+        # Given the desired link credit `link_credit`, the link credit sent via
+        # FlowFrame is calculated as follows: The link credit to flow on the wire
+        # `self.current_link_credit` is the desired link credit
+        # `link_credit` minus the current link credit on the wire `self.total_link_credit`.
+        self.current_link_credit = link_credit - self.total_link_credit if link_credit is not None \
```

Comment:
are there unit tests we can add to test flow credit math?

Reply:
I think so? will let this run out first

Reply:
added